mypy/fastparse.py

```diff
@@ -1042,7 +1040,7 @@ def transform_args(
         for a, kd in zip(args.kwonlyargs, args.kw_defaults):
             new_args.append(
                 self.make_argument(
-                    a, kd, ARG_NAMED if kd is None else ARG_NAMED_OPT, no_type_check
+                    a, kd, ARG_NAMED if kd is None else ARG_NAMED_OPT, no_type_check  # type: ignore[redundant-expr]
```

Comment:
This one does not look right: `kd` can be `None`, can't it?

Reply:
This isn't ast though. It's wrong in the `typed-ast` stubs: https://github.com/python/typeshed/blob/c0f27e39f9d424f9075b100f5c323f8422797ac9/stubs/typed-ast/typed_ast/ast3.pyi#L361

Reply:
Oh, of course! Mypy uses stdlib `ast` on 3.8+, but uses `typed-ast` for Python 3.7, and the selfcheck runs with `--python-version=3.7`.

Reply:
Typeshed PR here: https://github.com/python/typeshed/pull/8645.